pkg/standalone-utils/test/VaultPermit.test.ts

```diff
@@ -0,0 +1,70 @@
+import { ethers } from 'hardhat';
+import { Contract } from 'ethers';
+import { expect } from 'chai';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import Token from '@balancer-labs/v2-helpers/src/models/tokens/Token';
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+
+import { MAX_UINT256 } from '@balancer-labs/v2-helpers/src/constants';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { signPermit } from '@balancer-labs/v2-helpers/src/models/misc/signatures';
+
+const setup = async () => {
+  const [, admin, vault] = await ethers.getSigners();
+
+  const dai = await deploy('TestTokenPermitDAI', { args: [admin.address, 'DAI', 'DAI', 18] });
+  const mkr = await deploy('TestTokenPermit', { args: [admin.address, 'MKR', 'MKR', 18] });
+
+  const tokenList = [await Token.fromContract(dai), await Token.fromContract(mkr)];
+  const tokens = new TokenList(tokenList.sort((a, b) => a.compare(b)));
+
+  const vaultPermit = await deploy('VaultPermit', { args: [vault.address] });
+
+  return {
+    data: {},
+    contracts: {
+      tokens,
+      vaultPermit,
+    },
+  };
+};
+
+describe('VaultPermit', function () {
+  let tokens: TokenList, vaultPermit: Contract;
+  let admin: SignerWithAddress, vault: SignerWithAddress;
+
+  before('deploy base contracts', async () => {
+    [, admin, vault] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('set up asset manager', async () => {
+    const { contracts } = await setup();
+
+    tokens = contracts.tokens;
+    vaultPermit = contracts.vaultPermit;
+  });
+
+  describe('vaultPermit', () => {
+    it("approves the Vault to use the caller's tokens", async () => {
```

Comment:
If we reused a base layer ERC20Permit as mentioned on some of the comments above, we'd be able to rely on the tests for e.g. invalid nonce (which are currently BalancerPoolToken tests)